packages/js/src/executors/release-publish/release-publish.impl.ts

```diff
@@ -220,12 +221,54 @@ export default async function runExecutor(
       stdio: ['ignore', 'pipe', 'pipe'],
     });
 
-    const stdoutData = JSON.parse(output.toString());
+    /**
+     * We cannot JSON.parse the output directly because if the user is using lifecycle scripts, npm will mix its publish output with the JSON output all on stdout.
+     * Additionally, we want to capture and show the lifecycle script outputs as beforeJsonData and afterJsonData and print them accordingly below.
+     */
+    const extractJsonData = extractNpmPublishJsonData(output.toString());
+    if (!extractJsonData.jsonData) {
+      console.error(
+        'The npm publish output data could not be extracted. Please report this issue on https://github.com/nrwl/nx'
+      );
+      return {
+        success: false,
+      };
+    }
 
     // If npm workspaces are in use, the publish output will nest the data under the package name, so we normalize it first
-    const normalizedStdoutData = stdoutData[packageName] ?? stdoutData;
+    const normalizedStdoutData =
+      extractJsonData.jsonData[packageName] ?? extractJsonData.jsonData;
```

Comment:
I can't easily tell. Is the nested `packageName` specific form handled by the function above?

Can you ensure unit tests exist for both cases please?

Reply:
It’s not related to this PR at all

Reply:
I understand your question more now in the context of the question above about the regexp, I will clarify in the code

Reply:
Added a unit test for the nested format and updated the logic to ensure there is no spill over

Reply:
As part of my testing of npm to ensure that the required keys are present in all supported versions (and beyond), I (re)discovered the fundamental change that we made to this executor a little while ago.

In order to improve the npm config resolution (and in taking advice from Luke from NPM) we stopped running `npm publish` from the package root and began running it from the workspace root, pointing it to the package root instead.

Well it turns out the nesting behaviour that I handled on this line a long time ago (and the comment I wrote above it) are only _ever_ applicable when running with workspaces AND when executing the command from the package root. So this nesting never occurs in our executor any more and I therefore removed it entirely. This also lets us keep the new JSON extraction logic as simple as possible.

So that this knowledge is not lost I have added a comment above the npm publish command construction: https://github.com/nrwl/nx/pull/23850/commits/21b243d445a47becd8637debfebc9ba5b05ae476#diff-1c2ebfe42e321fc6f4a8c99988807e45610b304f4685123906741b0260fc5472R204-R211